skimage/filters/thresholding.py

```diff
@@ -1177,3 +1178,69 @@ def threshold_multiotsu(image, classes=3, nbins=256):
     thresh = bin_centers[thresh_idx]
 
     return thresh
+
+
+def threshold_phansalkar(image, window_size=15, k=0.25, r=None, p=2.0, q=10.0):
+    """Applies Phansalkar local threshold to an array. Phansalkar is a
+    modification of Sauvola technique to deal with low contrast images.
+
+    This method is using the following formula::
+        T = m(x,y) * (1 + p * exp( -q * m(x,y) ) + k * ((s(x,y) / R) - 1))
+    where m(x,y) and s(x,y) are the mean and standard deviation of
+    pixel (x,y) neighborhood defined by a rectangular window with size w
+    times w centered around the pixel. k, p and q are configurable parameters.
+    R is the maximum standard deviation of a greyscale image.
+
+    Parameters
+    ----------
+    image : ndarray
+        Input image.
+    window_size : int, or iterable of int, optional
+        Window size specified as a single odd integer (3, 5, 7, …),
+        or an iterable of length ``image.ndim`` containing only odd
+        integers (e.g. ``(1, 5, 5)``).
+    k : float, optional
+        Value of the positive parameter k.
+    r : float, optional
+        Value of R, the dynamic range of standard deviation.
+        If None, set to the half of the image dtype range.
+    p : float, optional
+        Value of the parameter p.
+    q : float, optional
+        Value of the parameter q.
+
+    Returns
+    -------
+    threshold : (N, M) ndarray
+        Threshold mask. All pixels with an intensity higher than
+        this value are assumed to be foreground.
+
+    Notes
+    -----
+    This algorithm is originally designed for detection of cell nuclei in low
+    contrast images. Therefore the historgram has to be equalized beforehand
+    using :func:`skimage.exposure.equalize_adapthist`.
+
+    References
+    ----------
+    .. [1] Phansalskar N. et al. "Adaptive local thresholding for detection of
+           nuclei in diversity stained cytology images.", International
+           Conference on Communications and Signal Processing (ICCSP),
+           pp. 218-220, 2011
+           :DOI:`10.1109/ICCSP.2011.5739305`
+
+    Examples
+    --------
+    >>> from skimage import data
+    >>> from skimage.exposure import equalize_adapthist
+    >>> image = data.moon()
+    >>> image_eq = equalize_adapthist(image)
+    >>> t_phansalkar = threshold_phansalkar(image_eq, window_size=15, k=0.25)
+    >>> binary_image = image_eq > t_phansalkar
+    """
+
+    if r is None:
+        imin, imax = dtype_limits(image, clip_negative=False)
+        r = 0.5 * (imax - imin)
```

Comment:
If the histogram equalization is integrated to the function, `imax - imin` can be replaced by `image.ptp()`

Reply:
Thank you for that suggestion.
I tried it for some images. And actually r will allways be 1, if the imput image is a result of the histogram equalization.
Therefore, I would suggest skipping the calculation and set `r = 1.0` in that case.

What is the advantage of image.ptp()? I tried to benchmark the function and it seems image.ptp() was a little slower on my system than the current code. But that is negligible in comparison to the thresholding itself.